### PR TITLE
[TA-1000] NearMobile: Bottom bar scan button

### PR DESCRIPTION
**Pull request by GuillemGarciaDev**
# [TA-1000] NearMobile: Bottom bar scan button

## Task

- [[TA-1000] NearMobile: Bottom bar scan button](https://www.notion.so/NearMobile-Bottom-bar-scan-button-1410e8242e6140d49fd04061f2270571?pvs=4)

## Changes

- `BottomBar` with QRCode button and dApps tab
- new `MainBottomBarItem` component (circle component in the middle of the BottomBar)
- New icons (Laptop & QRCode)
- new Button variant `contrast`
- `useThemeMode` hook util
- Stack updated with new routes
- testing updated for `BottomBar` and `MainBottomBarItem` components

### CODE REVIEW

File: src/module/common/component/input/Button/Button.styles.ts [hunk 1]
```diff
@@ -59,6 +63,9 @@ export const ButtonRoot = styled(Button)<ButtonProps>(({ theme, rounded = true }
             outlined: {
                 backgroundColor: theme.palette.overlay["8%"],
             },
+            contrast: {
```
Comment (JordiParraCrespo) [inline src/module/common/component/input/Button/Button.styles.ts:66]: Aquesta variant la utilitzes?

File: test/unit/src/module/common/component/navigation/BottomBar/BottomBar.spec.tsx [hunk 2]
```diff
@@ -38,6 +53,20 @@ describe("BottomBar test", () => {
         fireEvent.press(newsButton);
         expect(mockedNavigate).toHaveBeenCalledWith(MainBottomScreens.NEWS);
     });
+
+    test("Navigates to Dapps Screen", () => {
```
Comment (JordiParraCrespo) [inline test/unit/src/module/common/component/navigation/BottomBar/BottomBar.spec.tsx:57]: Aquí si fas servir el hook de `useConfig` i després el `useCOnfigMock` pots mockejar els feature flags

**Review by JordiParraCrespo — CHANGES_REQUESTED** [review]
Bona feina!! Mola el feauture flag aixi podem anar treient cosetes mentres!!

File: src/config/config.declarations.d.ts [hunk 3]
```diff
@@ -54,6 +54,7 @@ declare module "@peersyst/react-native-components" {
         nearMobileUrl: string;
         enableBuy: boolean;
         transak: TransakConfig;
+        signerFeature: boolean;
```
[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/config/config.declarations.d.ts:57]: Em mola molt la idea! Podriem migrar-ho a altres projectes on hi hagi una config una mica dispersa

[the remark this answers is not in the document]
Comment (GuillemGarciaDev) [reply src/config/config.declarations.d.ts:57]: Ho faig en una PR apart

File: src/module/common/component/input/Button/Button.styles.ts (hunk 1 above)
Comment (GuillemGarciaDev) [reply src/module/common/component/input/Button/Button.styles.ts:66]: Yes! Per al `MainBottomBarItem`

**Review by JordiParraCrespo — APPROVED** [review]

**Pull request merged**